### ISSUE TITLE
Support clickhouse

### ISSUE THREAD

**Issue by mgilbir** [EvgSkv/logica#75]
[Clickhouse](https://clickhouse.tech/) is a columnar store that can be queried with SQL.

**Comment by knoguchi** [comment]
It's interesting idea but Logica uses UDFs (user defined functions) which ClickHouse doesn't support.   I don't think it's feasible until ClickHouse implements UDFs.

**Comment by EvgSkv** [comment]
Logica compiles to UDFs only predicates defined with `-->` notation. It's not a huge burgen on the user willing to use Clickhouse not to use such notation. Furthermore, it's not too hard to turn off UDF compilation based on the engine. So I don't think this limitation is substantial.

**Comment by knoguchi** [comment]
Recent ClickHouse added UDF.  We may be able to use the new functionality while I understand UDF is not a substantial limitation.

**Comment by EvgSkv** [comment]
Yeah, that's right, UDFs were not  a limitation. I didn't work with Clickhouse, so I don't know what would be challenges.
Biggest challenge with full support of Postgres is that you can't use structs without defining their types. And Logic relies on records quite a bit. I don't know if this would be applicable to Clickhouse.
I'm happy to help with understanding Logica compiler to whomever wants to work on this :-)
